### PR TITLE
(WIP) Sponge support

### PR DESCRIPTION
**Pull request by SamB440**
Adds Sponge support. A lot is copied from the Spigot module as they are fundamentally similar and via runs on Sponge too.

The only conversion util I made is itemstack conversion. Most things can be done via Sponge API or VanillaGradle anyway.

~~Also fixed use of deprecated `java.sourceCompatibility` and `java.targetCompatibility` (need to confirm it compiles to 8 bytecode again).~~

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,8 +1,15 @@
+pluginManagement {
```
Comment (NoJokeFNA) [inline settings.gradle:1]: Why are you adding this into settings.gradle?

File: build.gradle [hunk 2]
```diff
@@ -26,8 +26,10 @@ subprojects {
                                        "net.kyori:adventure-text-serializer-legacy:${adventureVersion}",
                                        "net.kyori:adventure-nbt:${adventureVersion}"]
 
-    java.sourceCompatibility = JavaVersion.VERSION_17
-    java.targetCompatibility = JavaVersion.VERSION_1_8
+    java {
```
Comment (NoJokeFNA) [inline build.gradle:29]: I would prefer to have Gradle changes in separate PRs

File: settings.gradle (hunk 1 above)
Comment (SamB440) [reply settings.gradle]: I was messing with vanilla gradle at one point, gone now.

**Comment by SamB440** [comment]
Update for those unaware: viaversion support doesn't seem to be possible because on Sponge they replace the vanilla handler, instead of adding their own.

**Comment by SamB440** [comment]
I think I might migrate the current injection stuff to be mixin-based like fabric

**Comment by retrooper** [comment]
> Update for those unaware: viaversion support doesn't seem to be possible because on Sponge they replace the vanilla handler, instead of adding their own.

They have fixed this recently.

**Comment by retrooper** [comment]
You can mark this as a PR.

**Comment by retrooper** [comment]
Is this fully functional?

**Comment by SamB440** [comment]
Yes

**Comment by retrooper** [comment]
Thank you a lot for porting PacketEvents to Sponge!

**Comment by retrooper** [comment]
Next update, it should be listed as supported.

**Pull request merged**